code/datums/components/mood.dm

```diff
@@ -390,3 +390,16 @@
 	SIGNAL_HANDLER
 
 	setSanity(sanity + amount)
+
+/datum/component/mood/proc/HandleAddictions()
+	if(!iscarbon(parent))
+		return
+
+	var/mob/living/carbon/affected_carbon = parent
+
+	if(sanity < SANITY_GREAT) ///Sanity is low, stay addicted.
+		return
```

Comment:
We should really indicate to the player (somehow, at some point in the code) that they need to have good sanity in order to have their addictions go away.

Reply:
I mean, thats the whole crux of addictions that this system uses. The only real way to start dealing with your character's addiction is take measures to improve your mental health.

I don't think people really take moods seriously outside of movespeed. Thats somewhat of the secondary purpose of this pr, to make people take moods more seriously.

Perhaps something in medbay, or a status? Open to ideas